src/kv/kv_types.h

```diff
@@ -93,13 +93,55 @@ namespace kv
     Nodes nodes;
   };
 
+  inline void to_json(nlohmann::json& j, const Configuration::NodeInfo& ni)
+  {
+    j["address"] = fmt::format("{}:{}", ni.hostname, ni.port);
+  }
+
+  inline void from_json(const nlohmann::json& j, Configuration::NodeInfo& ni)
+  {
+    const std::string addr(j["address"]);
+    const auto& [h, p] = nonstd::split_1(addr, ":");
+    ni.hostname = h;
+    ni.port = p;
+  }
+
+  enum class ReplicaState
+  {
+    Leader,
+    Follower,
+    Candidate,
+    Retired
+  };
+
+  DECLARE_JSON_ENUM(
+    ReplicaState,
+    {{ReplicaState::Leader, "Leader"},
+     {ReplicaState::Follower, "Follower"},
+     {ReplicaState::Candidate, "Candidate"},
+     {ReplicaState::Retired, "Retired"}});
+
+  DECLARE_JSON_TYPE(Configuration);
+  DECLARE_JSON_REQUIRED_FIELDS(Configuration, idx, nodes);
```

Comment:
Its not obvious in the HTTP response what this `idx` is - is that the seqno where the configuration was created? Or the highest one it has acked?

Reply:
It's the creation/applicability seqno for the config.

Reply:
@eddyashton do you have a suggestion for a better name? I'd be happy to take it.

Reply:
`start_idx`, `valid_from`, `as_of`?